lib/rest-api.php

```diff
@@ -36,3 +36,23 @@ function gutenberg_register_edit_site_export_controller_endpoints() {
 	$edit_site_export_controller->register_routes();
 }
 add_action( 'rest_api_init', 'gutenberg_register_edit_site_export_controller_endpoints' );
+
+
+function gutenberg_register_archive_link_field() {
+	register_rest_field(
+		'type',
+		'archive_link',
+		array(
+			'get_callback'    => function ( $post_object ) {
```

Comment:
```suggestion
			'get_callback'    => static function ( $post_object ) {
```

Reply:
I realised that despite having `has_archive` as `false` the `post` archive ends up having a link set to the site's home url. 

```json
"post": {
        "description": "",
        "hierarchical": false,
        "has_archive": false, // this is correct 
        "name": "Posts",
        "slug": "post",
        "icon": "dashicons-admin-post",
        "taxonomies": [
            "category",
            "post_tag"
        ],
        "rest_base": "posts",
        "rest_namespace": "wp/v2",
        "template": [],
        "template_lock": false,
        "archive_link": "http://localhost:8888", // why does this happen????
}
```

No other object has this Issue so perhaps it's a bug. 

I corrected for this by filtering on the front end by `has_archive` rather than the presence of the `archive_link`.